engine/src/cmd/unit_csv.cpp

```diff
@@ -1048,87 +1047,80 @@ void Unit::LoadRow( CSVRow &row, string modification, string *netxml )
 
     // Get shield count
     int    shield_count = 0;
-    float shield_values[8];
+    float shield_values[4];
 
-    shield_values[0] = ::stof(OPTIM_GET( row, table, Shield_Front_Top_Left ));
-    shield_values[1] = ::stof(OPTIM_GET( row, table, Shield_Front_Top_Right ));
-    shield_values[2] = ::stof(OPTIM_GET( row, table, Shield_Back_Top_Left ));
-    shield_values[3] = ::stof(OPTIM_GET( row, table, Shield_Back_Top_Right ));
-    shield_values[4] = ::stof(OPTIM_GET( row, table, Shield_Front_Bottom_Left ));
-    shield_values[5] = ::stof(OPTIM_GET( row, table, Shield_Front_Bottom_Right ));
-    shield_values[6] = ::stof(OPTIM_GET( row, table, Shield_Back_Bottom_Left ));
-    shield_values[7] = ::stof(OPTIM_GET( row, table, Shield_Back_Bottom_Right ));
+    // TODO: this mapping should really go away
+    shield_values[0] = ::stof(OPTIM_GET( row, table, Shield_Front_Top_Right ));
+    shield_values[1] = ::stof(OPTIM_GET( row, table, Shield_Back_Top_Left ));
+    shield_values[2] = ::stof(OPTIM_GET( row, table, Shield_Front_Bottom_Right ));
+    shield_values[3] = ::stof(OPTIM_GET( row, table, Shield_Front_Bottom_Left ));
 
 
-    for(int i=0;i<8;i++) {
+    for(int i=0;i<4;i++) {
```

Comment:
perhaps `sizeof(shield_values)` (or something like that)  instead of `4`?
Ultimately we may want to change this to an iterable type so we could just use iterators:
```
for (auto sv = shield_values.begin(); sv != shield_values.end(); sv++) {
   if (*sv) {
       shield_count++;
   }
}
```

but it's probably not possible to make that change yet.